lib/logstash/inputs/tcp.rb

```diff
@@ -164,11 +171,14 @@ def handle_socket(socket, client_address, client_port, output_queue, codec)
       end
     end
   rescue EOFError
+    @metric_errors.increment(:connection)
```

Comment:
Should we consider EOF an error? EOF feels more like "The connection is done" and not really an error.